File: lib/features/profile/domain/entities/user_profile.dart
// User Profile Entity
class UserProfile {
  final String id;
  final String userId;
  final String name;
  final int age;
  final String gender;
  final double height; // in cm
  final double weight; // in kg
  final String? avatarUrl;

  UserProfile({
    required this.id,
    required this.userId,
    required this.name,
    required this.age,
    required this.gender,
    required this.height,
    required this.weight,
    this.avatarUrl,
  });
}

File: eater-backend/src/models/User_Profile.js
const mongoose = require("mongoose");

const User_ProfileSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Users",
      required: true,
    },
    firstName: { type: String, default: "" },
    lastName: { type: String, default: "" },
    phoneNumber: { type: String, default: "" },
    age: { type: Number, required: true },
    gender: { type: String, enum: ["male", "female", "other"], required: true },
    height: { type: Number, required: true }, // in centimeters
    weight: { type: Number, required: true }, // in kilograms
    goal_weight: { type: Number, default: null }, // in kilograms
    healthGoals: { type: String, default: "" }, // e.g., 'weight loss', 'muscle gain'
  },
  { timestamps: true },
);

const User_Profile = mongoose.model("User_Profiles", User_ProfileSchema);

module.exports = { User_Profile };
